### DISCUSSION TITLE
[FEATURE]建议对站点加上Allow traffic from Cloudflare only

### DISCUSSION THREAD

**Discussion by ossplus** [1Panel-dev/1Panel#4461] [Ideas]
### 1Panel 版本

lts

### 请描述您的需求或者改进建议

给现在的openResty网站增加一个选项：Allow traffic from Cloudflare only， 这对于在海外机器安装1panel的会比较有用。
![image](https://github.com/1Panel-dev/1Panel/assets/1935044/9ea5415d-bc6f-4d7c-8f56-0297b613e828)

**Comment by fred913** [comment]
建议参考[a post from ServerFault](https://serverfault.com/questions/1106260/how-to-configure-web-server-to-only-accept-traffic-from-cloudflare)
考虑到Answer #1也提到了，“Cloudflare never added an IP range in its history”，这个IP列表应当可以用于这个场景吧

可以试试这份配置
```nginx
location / {
    # v4
    allow 173.245.48.0/20;
    allow 103.21.244.0/22;
    allow 103.22.200.0/22;
    allow 103.31.4.0/22;
    allow 141.101.64.0/18;
    allow 108.162.192.0/18;
    allow 190.93.240.0/20;
    allow 188.114.96.0/20;
    allow 197.234.240.0/22;
    allow 198.41.128.0/17;
    allow 162.158.0.0/15;
    allow 104.16.0.0/13;
    allow 104.24.0.0/14;
    allow 172.64.0.0/13;
    allow 131.0.72.0/22;

    # v6
    allow 2400:cb00::/32;
    allow 2606:4700::/32;
    allow 2803:f800::/32;
    allow 2405:b500::/32;
    allow 2405:8100::/32;
    allow 2a06:98c0::/29;
    allow 2c0f:f248::/32;

    deny all;
}
```

**Comment by fred913** [comment]
考虑到1panel作为一个产品，感觉更适合引入的方案是 在防火墙（网站-安全）下引入一个“IP访问白名单”的功能，开启后添加IP段？然后Cloudflare的这些IP作为预设

**Comment by zhengkunwang223** [comment]
> 考虑到1panel作为一个产品，感觉更适合引入的方案是 在防火墙（网站-安全）下引入一个“IP访问白名单”的功能，开启后添加IP段？然后Cloudflare的这些IP作为预设

下个版本的 waf 支持 IP 黑白名单使用 IP 组
